cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -3361,3 +3361,22 @@
           (x64_andnpd sign_bit a)
           (x64_andpd sign_bit b))))
 
+;; Rules for `ceil` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type (use_sse41) (ceil a @ (value_type $F32))))
+      (x64_roundss a (RoundImm.RoundUp)))
+
+(rule (lower (ceil a @ (value_type $F32)))
```

Comment:
Here we're implicitly relying on the rule-ordering heuristic again (that pesky #4717!): this rule doesn't do anything to guard on *not* SSE4.1 and it's not clear to me that the sequence of `PatternInst`s is "less specific" and shadowed by the above rule without going through the compilation in my head... let's leave this for now though as we are likely to resolve #4717 soon.